### ISSUE TITLE
(lambda): Not able to update imported lambda role using the fromFunctionAttributes lambda importer

### ISSUE THREAD

**Issue by Spaceage1217** [aws/aws-cdk#19566]
### What is the problem?

Trying to update a imported lambda role in order to add the following permissions  `"sqs:ReceiveMessage",
          "sqs:ChangeMessageVisibility",
          "sqs:GetQueueUrl",
          "sqs:DeleteMessage",
          "sqs:GetQueueAttributes", ` to the lambda via a role.
          
     ` const myLambda = lambda.Function.fromFunctionAttributes(
          this,
          "myLambda",
          {
            functionArn: myLambdaArn,
            role: new iam.Role(this, "myLambdaRole", {
              assumedBy: new iam.ServicePrincipal("lambda.amazonaws.com"),
              inlinePolicies: {
                x: new iam.PolicyDocument({
                  statements: [
                    new iam.PolicyStatement({
                      actions: [
                        "sqs:ReceiveMessage",
                        "sqs:ChangeMessageVisibility",
                        "sqs:GetQueueUrl",
                        "sqs:DeleteMessage",
                        "sqs:GetQueueAttributes",
                      ],
                      resources: [myQueue.queueArn],
                    }),
                  ],
                }),
              },
            }),
            sameEnvironment: true,
          }
        );
        
        ....

       myLambda.addEventSource(
            new lambdaEventSources.SqsEventSource(myQueue)
          );
        
        `

The imported lambda, was created with Amplify.

### Reproduction Steps

1. Import a already created lambda in the same account but seprate stack.
2. Create a queue with CDK
3. use addEventSource to create a lambda event source, mapping the lambda to the queue.

### What did you expect to happen?

Ideally, what I expected was that using `addEventSource` would create the role already, and although, I am able to see the role being created (from addEventSource) in the cdk output, but when I go to build in cloud, I see the following error `Resource handler returned message: "Invalid request provided: The provided execution role does not have permissions to call ReceiveMessage on SQS`

### What actually happened?

 I see the following error `Resource handler returned message: "Invalid request provided: The provided execution role does not have permissions to call ReceiveMessage on SQS`

### CDK CLI Version

1.124

### Node.js Version

v14.17.4

### OS

Windows

### Language

Typescript

### Language Version

4.2.4

### Other information

The imported lambda, was created with Amplify.

**Comment by ryparker** [comment]
Hey @Spaceage1217 👋🏻  Thanks for opening the issue. 

The reason you're seeing this error is because CDK does not support updating an imported resource.

> Although you can use an imported resource anywhere, you cannot modify the imported resource. For example, calling addToResourcePolicy (Python: add_to_resource_policy) on an imported s3.Bucket does nothing.

[_importing existing external resources docs_](https://docs.aws.amazon.com/cdk/v2/guide/resources.html#resources_importing)

Here are a couple alternative options:
- You could update the resource using the AWS SDK, console, CLI, etc. 
- Create the resource using CDK.

**Comment by Spaceage1217** [comment]
Oh ok that’s interesting, any reason why it fails silently instead of issuing a warning? Also, is it supposed to be possible to update policies (not roles) of a imported lambda? I’ve done it before which is why I assumed roles would act the same way. I was able to give a imported lambda permissions so a bucket I created in cdk could use it as a trigger, although I had to go to the lower level constructs and use CFN.
